resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventOutputImpl.java

```diff
@@ -105,10 +110,48 @@ private void internalFlushResponseToClient(boolean throwIOException) throws IOEx
             {
                jaxrsResponse = (BuiltResponse) Response.noContent().build();
             }
-            else
+            else //set back to client 200 OK to implies the SseEventOutput is ready
             {
-               //set back to client 200 OK to implies the SseEventOutput is ready
-               jaxrsResponse = (BuiltResponse) Response.ok().type(MediaType.SERVER_SENT_EVENTS).build();
+               ResourceMethodInvoker method =(ResourceMethodInvoker) request.getAttribute(ResourceMethodInvoker.class.getName());
+               Produces produces = method.getMethod().getAnnotation(Produces.class);
+               if (produces != null & contains(produces.value(), MediaType.SERVER_SENT_EVENTS))
```

Comment:
Typo? '&' -> '&&'?

Reply:
@marekkopecky, you're right again. Oddly, I think the result is the same even though the semantics are different.